drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationGridHeaderUtilities.java

```diff
@@ -90,10 +90,29 @@ public static ScenarioHeaderMetaData getColumnScenarioHeaderMetaData(final Scena
         return ci.getColumn();
     }
 
-    public static boolean isEditableHeader(final GridColumn<?> column,
+    /**
+     * Checks whether the edit mode can be invoked on header cell from given column on given row.
+     * @param column
+     * @param uiHeaderRowIndex
+     * @return true if conditions are met, false otherwise
+     */
+    public static boolean isEditableHeader(final ScenarioGridColumn column,
                                            final Integer uiHeaderRowIndex) {
-        GridColumn.HeaderMetaData headerMetaData = column.getHeaderMetaData().get(uiHeaderRowIndex);
-        return headerMetaData instanceof ScenarioHeaderMetaData && !((ScenarioHeaderMetaData) headerMetaData).isReadOnly();
+        final GridColumn.HeaderMetaData headerMetaData = column.getHeaderMetaData().get(uiHeaderRowIndex);
+        if (!(headerMetaData instanceof ScenarioHeaderMetaData)) {
```

Comment:
@jomarko 
In this case, an exception must be thrown

Reply:
@gitgabrio thank you, I pushed required changes